core/src/test/scala/integration/kafka/zk/ZkMigrationIntegrationTest.scala

```diff
@@ -324,4 +329,29 @@ class ZkMigrationIntegrationTest {
       assertTrue(firstProducerIdBlock.firstProducerId() < producerIdBlock.firstProducerId())
     }
   }
+
+  /**
+   * Start a KRaft cluster with migrations enabled, verify that the controller does not accept metadata changes
+   * through the RPCs
+   */
+  @ClusterTests(Array(
+    new ClusterTest(clusterType = Type.KRAFT, brokers = 0, controllers = 1, metadataVersion = MetadataVersion.IBP_3_4_IV0,
+      serverProperties = Array(new ClusterConfigProperty(key = "zookeeper.metadata.migration.enable", value = "true"))),
+    new ClusterTest(clusterType = Type.KRAFT, brokers = 0, controllers = 1, metadataVersion = MetadataVersion.IBP_3_5_IV0,
+      serverProperties = Array(new ClusterConfigProperty(key = "zookeeper.metadata.migration.enable", value = "true"))),
+    new ClusterTest(clusterType = Type.KRAFT, brokers = 0, controllers = 1, metadataVersion = MetadataVersion.IBP_3_5_IV1,
+      serverProperties = Array(new ClusterConfigProperty(key = "zookeeper.metadata.migration.enable", value = "true")))
+  ))
+  def testPreMigrationMode(clusterInstance: ClusterInstance): Unit = {
```

Comment:
Hmm... I don't see how this can work without brokers, since the admin client needs to bootstrap through broker servers, right? So I don't think this is testing what you might think.

You might just have to send a raw RPC, as unpleasant as that is to do. That will also allow you to verify that you get NOT_CONTROLLER.

Reply:
Heh, the test framework is politely adding a broker for me :)

This test does seem to do what I want, i see:

```
[2023-04-20 17:56:08,267] ERROR [QuorumController id=3000] createTopics: unable to start processing because of NotControllerException. Reason: The active controller appears to be node 3000 (org.apache.kafka.controller.QuorumController:445)
```

in the log, and then the create topics call times out after a short time. I've updated the test to have `brokers = 1`

Reply:
Thanks.

```
Reason: The active controller appears to be node 3000
```

can you change it so that this talks about premigration?